docs/dogstatsd/internals.md

```diff
@@ -0,0 +1,53 @@
+![Dogstatsd Internals Schema](internals_schema.png)
+
+## Packet
+
+In Dogstatsd, a Packet is a bytes array containing one or multiple metrics in the Dogstatsd format (separated by a \n when there are several). Its maximum size is dogstatsd_buffer_size.
+
+## PacketAssembler
+
+Input: a datagram from an UDP socket
+Output: a Packet containing multiple metrics packed together, separated by a \n
+
+The PacketAssembler is gathering multiple datagrams into one Packet of maximum size dogstatsd_buffer_size sending it to the PacketsBuffer, avoiding to run the whole parsing pipeline with only 1 metric per packet. The bytes buffer used comes from the PacketPool, avoiding having to re-allocate bytes buffer every time. Please note that the UDS pipeline does not use the PacketAssembler because each UDS packet also contains metadata (origin tags) which are used to enrich the metrics tags, making them impossible to be packed together by the PacketAssembler.
```

Comment:
```suggestion
The PacketAssembler gathers multiple datagrams into one Packet of maximum size, `dogstatsd_buffer_size`, and sends it to the PacketsBuffer which avoids running the whole parsing pipeline with only one metric per packet. The bytes buffer used comes from the PacketPool, which avoids re-allocating the bytes buffer every time. 

**Note**: The UDS pipeline does not use the PacketAssembler because each UDS packet also contains metadata (origin tags) which are used to enrich the metrics tags, making them impossible to be packed together by the PacketAssembler.
```